lib/js/src/manager/screen/utils/_PresentAlertOperation.js

```diff
@@ -0,0 +1,492 @@
+     * @param {AlertView} alertView - An instance of AlertView.
+     * @param {WindowCapability} currentCapabilities - The capabilities of the default main window.
+     * @param {SpeechCapabilities[]|null} speechCapabilities - An array of SpeechCapabilities enums or null.
+     * @param {FileManager} fileManager - An instance of FileManager.
+     * @param {Number} cancelId - The ID to be used during CancelInteraction.
+     * @param {Function} listener - Callback function that is used when operation has finished.
+     */
+    constructor (lifecycleManager, alertView, currentCapabilities, speechCapabilities, fileManager = null, cancelId, listener) {
+        super();
+        this._lifecycleManager = lifecycleManager;
+        this._alertView = alertView.clone();
+        this._currentWindowCapability = currentCapabilities;
+        this._speechCapabilities = speechCapabilities;
+        this._fileManager = fileManager;
+        this._cancelId = cancelId;
+        this._listener = listener;
+        this._isAlertPresented = false;
+
+        this._alertView.canceledListener = function () {
+            this.cancelAlert();
+        };
+    }
+
+    /**
+     * The method that causes the task to run.
+     * @param {_Task} task - The task instance
+     */
+    onExecute (task) {
+        this._start();
+    }
+
+    /**
+     * If the task is not canceled, starts to assemble the alert
+     * @private
+     */
+    async _start () {
+        if (this.getState() === _Task.CANCELED) {
+            this._finishOperation(false);
+            return;
+        }
+
+        if (!this.isValidAlertViewData(this._alertView)) {
+            if (this._alertView.getAudio() !== null && this._alertView.getAudio() !== undefined &&
+                this._alertView.getAudio().getAudioData().length > 0) {
+                console.log('The module does not support the use of only audio file data in an alert. ' +
+                'The alert has no data and can not be sent to the module. ' +
+                        'The use of audio file data in an alert is only supported on modules supporting RPC Spec v5.0 or newer');
+            } else {
+                console.log('The alert data is invalid.' +
+                ' At least either text, secondaryText or audio needs to be provided. ' +
+                        'Make sure to set at least the text, secondaryText or audio properties on the AlertView');
+            }
+            this._finishOperation(false, null);
+            return;
+        }
+        const uploadFilesTask = new _DispatchGroup();
+
+        // Enter DispatchGroup twice for two tasks needing to be completed, One for uploading images and one for uploading audio files
+        uploadFilesTask.enter();
+        uploadFilesTask.enter();
+
+        // DispatchGroup notify when all tasks are done
+        uploadFilesTask.notify(async () => {
+            await this.presentAlert();
```

Comment:
I wrapped this logic in a Promise like you did, and also converted the uploadImages and uploadAudioFiles methods to be passed listeners that will then call `leave()` instead of using the await syntax.